lib/drivers/tandemTslimDriver.js

```diff
@@ -1328,45 +1328,55 @@ module.exports = function (config) {
 
   var buildBasalRecords = function (data, records) {
     var basalRecords = filterLogEntries(
-      [PUMP_LOG_RECORDS.LID_BASAL_RATE_CHANGE],
+      [PUMP_LOG_RECORDS.LID_BASAL_RATE_CHANGE,PUMP_LOG_RECORDS.LID_NEW_DAY],
```

Comment:
Not a fan of this approach - will explain why tonight. We should have a separate function `buildNewDayRecords` and simulate new day records separately.